src/neo4j_genai/retrievers/base.py

```diff
@@ -81,8 +81,9 @@ class Retriever(ABC, metaclass=RetrieverMetaclass):
     index_name: str
     VERIFY_NEO4J_VERSION = True
 
-    def __init__(self, driver: neo4j.Driver):
+    def __init__(self, driver: neo4j.Driver, database: Optional[str] = None):
```

Comment:
(Not related to this PR): In terms of data validation, what about validating driver and database here, instead of in each retriever' model?

I mean having something like:

```
class RetrieverInitModel(BaseModel):
   driver: neo4j.Driver # (or Any I don't rememberà
   database: str
   # usual validation of these

class Retriever:
   def __init__(driver, database):
      try:
         m = RetrieverInitModel(driver, database)
      except ValidationError:
         raise TheCorrectException

class VectorRetriever(Retriever):
    def __init__(driver, index_name, database):
       super().__init__(driver, database)  # driver and database validation happens here
       # then we can validate the rest of the inputs
```

Reply:
That would help with not duplicating code across retrievers, how about that implement in a future PR?